### ISSUE TITLE
fix indentation on multi-line map literals in named args

### ISSUE THREAD

**Issue by sigmundch** [dart-lang/dart_style#162]
For example, this code:

```dart
library app_bootstrap;

import 'package:polymer/polymer.dart';

import 'package:polymer/src/build/log_injector.dart';
import 'sort_registration_test.dart' as i0;
import 'package:smoke/smoke.dart' show Declaration, PROPERTY, METHOD;
import 'package:smoke/static.dart' show useGeneratedCode, StaticConfiguration;
import 'sort_registration_test.dart' as smoke_0;
import 'package:polymer/polymer.dart' as smoke_1;

void main() {
  useGeneratedCode(new StaticConfiguration(
      checkedMode: false,
      parents: {
        smoke_0.A: smoke_1.PolymerElement,
        smoke_0.B: smoke_0.A,
        smoke_0.C: smoke_0.B,
        smoke_0.D: smoke_0.B,
        smoke_0.E: smoke_0.C,
      },
      declarations: {
        smoke_0.A: {},
        smoke_0.B: {},
        smoke_0.C: {},
        smoke_0.D: {},
        smoke_0.E: {},
      }));
  new LogInjector().injectLogsFromUrl('sort_registration_test.html._buildLogs');
  configureForDeployment([
      () => Polymer.register('x-a', i0.A),
      () => Polymer.register('x-b', i0.B),
      () => Polymer.register('x-d', i0.D),
      () => Polymer.register('x-c', i0.C),
      () => Polymer.register('x-e', i0.E),
    ]);
  i0.main();
}
```

is currently is formatted as:

```dart
library app_bootstrap;

import 'package:polymer/polymer.dart';

import 'package:polymer/src/build/log_injector.dart';
import 'sort_registration_test.dart' as i0;
import 'package:smoke/smoke.dart' show Declaration, PROPERTY, METHOD;
import 'package:smoke/static.dart' show useGeneratedCode, StaticConfiguration;
import 'sort_registration_test.dart' as smoke_0;
import 'package:polymer/polymer.dart' as smoke_1;

void main() {
  useGeneratedCode(new StaticConfiguration(
      checkedMode: false,
      parents: {
    smoke_0.A: smoke_1.PolymerElement,
    smoke_0.B: smoke_0.A,
    smoke_0.C: smoke_0.B,
    smoke_0.D: smoke_0.B,
    smoke_0.E: smoke_0.C,
  },
      declarations: {
    smoke_0.A: {},
    smoke_0.B: {},
    smoke_0.C: {},
    smoke_0.D: {},
    smoke_0.E: {},
  }));
  new LogInjector().injectLogsFromUrl('sort_registration_test.html._buildLogs');
  configureForDeployment([
    () => Polymer.register('x-a', i0.A),
    () => Polymer.register('x-b', i0.B),
    () => Polymer.register('x-d', i0.D),
    () => Polymer.register('x-c', i0.C),
    () => Polymer.register('x-e', i0.E),
  ]);
  i0.main();
}
```

I'd expect the body of the maps to be indented + 2 from the name argument indentation

**Comment by sigmundch** [comment]
This is similar to other issues about how to nest literals and closures (see #24 and #145), but deserves its own issue, because, well, the style guide could be uplifted on this regard ;-)

@alan-knight agrees too :)

**Comment by zoechi** [comment]
Another example

```Dart
    final server = new DatastoreLocalDevServer('connect',
        workingDirectory: path.join(
            packageRoot().absolute.path, 'test/tmp_data'),
        environment: <String, String>{
      'JAVA': '/usr/lib/jvm/java-7-openjdk-amd64/bin/java'
    },
        minUpTimeBeforeShutdown: new Duration(seconds: 0));
```
should be
```Dart
    final server = new DatastoreLocalDevServer('connect',
        workingDirectory: path.join(
            packageRoot().absolute.path, 'test/tmp_data'),
        environment: <String, String>{
            'JAVA': '/usr/lib/jvm/java-7-openjdk-amd64/bin/java'},
        minUpTimeBeforeShutdown: new Duration(seconds: 0));
```
or 

```Dart
    final server = new DatastoreLocalDevServer('connect',
        workingDirectory: path.join(
            packageRoot().absolute.path, 'test/tmp_data'),
        environment: <String, String>{
            'JAVA': '/usr/lib/jvm/java-7-openjdk-amd64/bin/java'
        },
        minUpTimeBeforeShutdown: new Duration(seconds: 0));
```

**Comment by munificent** [comment]
Fixed by 346ad0736b5aff21ce5751c818bd4ccfc02d5885!
